src/xcode/ENA/ENA/Source/Scenes/ContactDiary/DiaryService.swift

```diff
@@ -55,97 +72,117 @@ class MockDiaryStore: DiaryStoring {
+	func addContactPersonEncounter(contactPersonId: Int64, date: String) -> DiaryStoringResult {
 		let id = contactPersonEncounters.map { $0.id }.max() ?? -1 + 1
 		contactPersonEncounters.append(ContactPersonEncounter(id: id, date: date, contactPersonId: contactPersonId))
 
 		updateDays()
 
-		return id
+		return .success(id)
 	}
 
 	@discardableResult
-	func addLocationVisit(locationId: Int, date: String) -> Int {
+	func addLocationVisit(locationId: Int64, date: String) -> DiaryStoringResult {
 		let id = locationVisits.map { $0.id }.max() ?? -1 + 1
 		locationVisits.append(LocationVisit(id: id, date: date, locationId: locationId))
 
 		updateDays()
 
-		return id
+		return .success(id)
 	}
 
-	func updateContactPerson(id: Int, name: String) {
-		guard let index = contactPersons.firstIndex(where: { $0.id == id }) else { return }
+	func updateContactPerson(id: Int64, name: String) -> DiaryStoringVoidResult {
+		guard let index = contactPersons.firstIndex(where: { $0.id == id }) else { return .success(()) }
 		contactPersons[index] = DiaryContactPerson(id: id, name: name)
 
 		updateDays()
+
+		return .success(())
 	}
 
-	func updateLocation(id: Int, name: String) {
-		guard let index = locations.firstIndex(where: { $0.id == id }) else { return }
+	func updateLocation(id: Int64, name: String) -> DiaryStoringVoidResult {
+		guard let index = locations.firstIndex(where: { $0.id == id }) else { return .success(()) }
 		locations[index] = DiaryLocation(id: id, name: name)
 
 		updateDays()
+
+		return .success(())
 	}
 
-	func removeContactPerson(id: Int) {
+	func removeContactPerson(id: Int64) -> DiaryStoringVoidResult {
 		contactPersons.removeAll { $0.id == id }
 		contactPersonEncounters.removeAll { $0.contactPersonId == id }
 
 		updateDays()
+
+		return .success(())
 	}
 
-	func removeLocation(id: Int) {
+	func removeLocation(id: Int64) -> DiaryStoringVoidResult {
 		locations.removeAll { $0.id == id }
 		locationVisits.removeAll { $0.locationId == id }
 
 		updateDays()
+
+		return .success(())
 	}
 
-	func removeContactPersonEncounter(id: Int) {
+	func removeContactPersonEncounter(id: Int64) -> DiaryStoringVoidResult {
 		contactPersonEncounters.removeAll { $0.id == id }
 
 		updateDays()
+
+		return .success(())
 	}
 
-	func removeLocationVisit(id: Int) {
+	func removeLocationVisit(id: Int64) -> DiaryStoringVoidResult {
 		locationVisits.removeAll { $0.id == id }
 
 		updateDays()
+
+		return .success(())
 	}
 
-	func removeAllLocations() {
+	func removeAllLocations() -> DiaryStoringVoidResult {
 		locations.removeAll()
 		locationVisits.removeAll()
 
 		updateDays()
+
+		return .success(())
 	}
 
-	func removeAllContactPersons() {
+	func removeAllContactPersons() -> DiaryStoringVoidResult {
 		contactPersons.removeAll()
 		contactPersonEncounters.removeAll()
 
 		updateDays()
+
+		return .success(())
+	}
+
+	func cleanup() -> DiaryStoringVoidResult {
+		return .success(())
```

Comment:
I assume this is for the moment, right? Could you add a comment that this is work in progress (in case we miss this later)

Reply:
This is the mock implementation.
I don't know if a cleanup makes sense for the mockup.
Nevertheless, I have added a comment.